flutter/lib/src/view_hierarchy/view_hierarchy_event_processor.dart

```diff
@@ -0,0 +1,29 @@
+import 'dart:async';
+
+import '../../sentry_flutter.dart';
+import '../binding_utils.dart';
+import 'sentry_tree_walker.dart';
+
+class SentryViewHierarchyEventProcessor implements EventProcessor {
+  @override
+  FutureOr<SentryEvent?> apply(SentryEvent event, {Hint? hint}) async {
+    if (event.exceptions == null && event.throwable == null) {
+      return event;
+    }
+
+    final instance = BindingUtils.getWidgetsBindingInstance();
+    if (instance == null) {
+      return event;
+    }
+    final sentryViewHierarchy = walkWidgetTree(instance);
```

Comment:
Dart is single-threaded, so no multi-threading concern, its also a sync call.
Running on debug mode (Profile mode isn't supported on iOS simulator, Debug is much slower) takes ~`30ms`, for our sample app/main screen, this is dynamic because it depends on the app's view, so hard to measure.
Final JSON around ~50-60kb

Reply:
> if there are any multi-threading concerns (e.g. walking the tree whilst it get's modified)

I actually think that's a valid concern because that's also a common problem when creating screenshots

Reply:
The feature is best effort, if for some reason you can't read the tree or take the screenshot, that's fine, there will be always a state that the framework won't allow you to do it, and we don't have control over that.